### ISSUE TITLE
Some PHP classes are not properly extracted/documented since 1.8.16

### ISSUE THREAD

**Issue by jtraulle** [doxygen/doxygen#7243]
Hello,

I have noticed that since Doxygen 1.8.16, some classes of a PHP project that I am trying to generate with Doxygen are not documented anymore.

I saw that Doxygen 1.8.16 fixes lots of issues with the LaTeX / PDF generation and it would be great to use this new version to be able to generate an offline doc in PDF format but this particular issue is a blocker.

The issue is not consistent : some classes gets correctly generated, some do not.

Any help would be appreciated :+1: :wink: 

### Screenshot working on Doxygen 1.8.13
_I have tested also with **1.8.15** and it is still working._

![Screenshot_2019-09-02 Dolibarr CommonObject Class Reference](https://user-images.githubusercontent.com/613615/64113619-164ffd80-cd8b-11e9-8b75-cb6f9bd99438.png)

![Screenshot_2019-09-02 Dolibarr CommonObject Class Reference(1)](https://user-images.githubusercontent.com/613615/64113630-1b14b180-cd8b-11e9-9879-5d43f108273d.png)

### Screenshot broken on Doxygen 1.8.16

![Screenshot_2019-09-02 Dolibarr CommonObject Class Reference](https://user-images.githubusercontent.com/613615/64113655-2cf65480-cd8b-11e9-8e4a-e84417982163.png)

I have also noticed that the file mention (The documentation for this class was generated from the following file:) is wrong on 1.8.16 ; it should be **htdocs/core/class/commonobject.class.php** but Doxygen output **htdocs/core/class/link.class.php**

## Minimal reproducible example

1. Into an new empty directory, clone the develop branch of the Dolibarr project.

```bash
git clone https://github.com/Dolibarr/dolibarr.git -b develop --single-branch --depth 1 dolibarr
```

2. Use this Doxygen config file (to rename `dolibarr.doxyfile`)

* [dolibarr.doxyfile.txt](https://github.com/doxygen/doxygen/files/3565986/dolibarr.doxyfile.txt)

3. Run with

```bash
( cat dolibarr.doxyfile ; echo 'PROJECT_NUMBER=develop' ) | doxygen -
```

**Comment by albert-github** [comment]
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

The mentioned develop branch of the Dolibarr project is a moving target and as such not really suitable to debug the problem.

**Comment by jtraulle** [comment]
Thanks for your reply @albert-github 

Here is a self contained example :
- [dolibarr-doxygen.tar.gz](https://github.com/doxygen/doxygen/files/3566110/dolibarr.tar.gz)

I have truncated the folder tree to reduce the size of the archive but the issue is still reproducible and can be seen on the **CommonObject Class Reference** Doxygen generated page.

**Comment by albert-github** [comment]
As far as I can see the difference is caused by commit: 61a987f886200cc62a786c3bd558f428062cf529 (issue 7068: "QGDict::hashAsciiKey: Invalid null key" error messages (part 2))

Some notes on the provided exampple:
- minimal looks a little bit different to me ...
- the issue is a blocker for you, but I think there are also quite a few warnings that should be resolved (and can easily be resolved).

**Comment by jtraulle** [comment]
I am sorry :confused: Indeed, I skipped the **minimal** part ...

----
Here is a really minimal example (only two PHP files + doxygen config file) :
- [dolibarr-doxygen-minimal.tar.gz](https://github.com/doxygen/doxygen/files/3566745/dolibarr.tar.gz) (91.8 Ko)

On this example, the class `CommonInvoice` is correctly generated and the class `CommonObject` is not.

----

This minimal example generates without any warning. Thanks for pointing the warnings out. The code base is relatively old and fixing that is an ongoing effort (most of the warning are related to incorrect syntax for documenting the class attributes (variables). I will probably take care of that but it will take some time to review (but I agree this should be pretty easy).

----

I have never compiled Doxygen before (I am currently using the distro packages on Arch Linux and Alpine Linux) but I will try to clone the `Release_1_8_16` tag, revert the mentioned commit and compile to see if this helps :slightly_smiling_face: I will let you know !

Thanks for taking the time to help :+1:

**Comment by doxygen** [comment]
@albert-github commit https://github.com/doxygen/doxygen/commit/61a987f886200cc62a786c3bd558f428062cf529 introduced a rule `<*>"?>"|"</script>"` on line 6907 and the php script used as input, has a `</script>` inside a string literal causing the rest of the file to be ignored. So this is another example of matching something too liberally. We really need to be much more careful with this...

**Comment by doxygen** [comment]
I've pushed [this commit](https://github.com/doxygen/doxygen/commit/4fe4ce026d75972018516d81507ae20db9cdc589) to address the issue. Maybe it is now too strict but we can always make it less strict if there is a valid example that requires it.

**Comment by jtraulle** [comment]
Great ! Thanks @doxygen :wink: 
I have checkout out the `Release_1_8_16` tag and cherry picked your commit and this is fixed :+1:

**Comment by albert-github** [comment]
I reran the test cases as indicated in #7068 and none of the cases gave "QGDict::hashAsciiKey: Invalid null key".

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
